win_etw_provider/src/guid.rs

```diff
@@ -122,6 +122,15 @@ impl From<uuid::Uuid> for GUID {
     }
 }
 
+#[cfg(feature = "uuid")]
+impl From<&str> for GUID {
```

Comment:
This needs to be TryFrom.  From is for conversions that cannot fail.

If you want to have const constructors for GUID, then we need a different way to handle that.  But From impls are expected not to panic.